overlord/snapstate/catalogrefresh.go

```diff
@@ -94,9 +97,24 @@ func refreshCatalogs(st *state.State, theStore StoreService) error {
 		return err
 	}
 	defer namesFile.Cancel()
-	if err := theStore.WriteCatalogs(namesFile); err != nil {
+
+	cmdDB, err := newCmdDB()
+	if err != nil {
+		return err
+	}
+
+	defer cmdDB.Rollback()
```

Comment:
Hm, I find this slightly confusing to have an unconditional rollback here? maybe a comment at least what is going on?